sql/atxs/atxs.go

```diff
@@ -208,6 +208,30 @@ func GetIDsByEpoch(db sql.Executor, epoch types.EpochID) (ids []types.ATXID, err
 	return ids, nil
 }
 
+// VRFNonce gets the VRF nonce of a smesher for a given epoch.
+func VRFNonce(db sql.Executor, id types.NodeID, epoch types.EpochID) (nonce types.VRFPostIndex, err error) {
+	enc := func(stmt *sql.Statement) {
+		stmt.BindBytes(1, id.Bytes())
+		stmt.BindInt64(2, int64(epoch))
+	}
+	dec := func(stmt *sql.Statement) bool {
+		nonce = types.VRFPostIndex(stmt.ColumnInt64(0))
+		return true
+	}
+
+	if rows, err := db.Exec(`
+		select nonce from atxs
+		where smesher = ?1 and epoch <= ?2 and nonce is not null
```

Comment:
the recorded epoch is publish epoch. so here i think you want `<` instead of `<=`

Reply:
never mind. i am wrong.

Reply:
actually you do want `<` here. the epoch you passed in is target epoch

Reply:
You are probably right: when I wrote that query I was persisting the nonce based on `TargetEpoch`, but `epoch` here refers to `PublishEpoch`. Changed `<=` to `<`.